### PR TITLE
Code refactoring

### PR DESCRIPTION
**Pull request by zombieJ**
* Use context api for passing function props (HOC prepared popus)
* Show warning in console instead of crash when has non-TreeNode child
* Fix switcher logic
* Tree level disabled logic support
* Fix check logic (ignore disabled child | parent)
* Optimize check logic (Conduct check by path instead of Tree trevarse, this affect the order of checkedKeys in callback)
* If loadData with expanded key, auto load it
* Fix dynamic node change disabled props
* Adjust default props logic (coontrolled component is always cover the default)
* Fix onDrop display logic when Drag out of the tree
* Common logic now move into `util.js`

### CODE REVIEW

**Comment by zombieJ** [comment]
https://github.com/ant-design/ant-design/issues/9531

**Comment by zombieJ** [comment]
https://github.com/ant-design/ant-design/issues/9304

File: src/util.js [hunk 1]
```diff
@@ -1,6 +1,29 @@
...
+  const index = clone.indexOf(value);
+  if (index >= 0) {
+    clone.splice(index, 1);
+  }
+  return clone;
+}
+
+export function arrAdd(list, value) {
+  const clone = list.slice();
+  if (!clone.includes(value)) {
```
Comment (yesmeck) [inline src/util.js]: 用 `indexOf` 吧

Comment (zombieJ) [reply src/util.js]: done

**Pull request merged**